routes/api/user.js

```diff
@@ -15,27 +17,58 @@ router.post('/register', async function(req, res, next) {
     email: req.body.email,
     password: hash
   }
-  const existingUser = await userModel.findOne({ $or: [{email:req.body.email, username:req.body.username}]})
-  if(existingUser) {
-    res.status(401).send({message:"Email or Username should be unique"})
-  } else {
-    await userModel.create(user)
-    res.send({message: "User registered Successfully"})
+  const existingUser = await userModel.findOne({ $or: [{email:{$eq: req.body.email}}, {username:{$eq: req.body.username}}]})
+  if(!existingUser) {
+    const registerUser = await userModel.create(user)
+    const transporter = nodemailer.createTransport({
+      host: "smtp.gmail.com",
```

Comment:
I have heard using Gmail SMTP is problematic.

May I know why you're planning to use Gmail SMTP server?

Reply:
For now, this works. please suggest some alternative which can be implemented later

Reply:
Need to change this later.